File: mainActivity/src/main/java/net/airvantage/utils/AvPhonePrefs.java
package net.airvantage.utils;

public class AvPhonePrefs {

    public String serverHost;
    public String clientId;

    public String password;
    public String period;

    public PreferenceUtils.Server usesServer;

    public boolean checkCredentials() {
        return !(password == null || password.isEmpty() || serverHost == null || serverHost.isEmpty());
    }

    public boolean usesNA() {
        return usesServer == PreferenceUtils.Server.NA;
    }

    public boolean usesEU() {
        return usesServer == PreferenceUtils.Server.EU;
    }

    public boolean usesCustomServer() {
        return usesServer == PreferenceUtils.Server.CUSTOM;
    }

}

File: mainActivity/src/main/java/net/airvantage/utils/AvPhonePrefs.kt
package net.airvantage.utils

class AvPhonePrefs {

    var serverHost: String? = null
    var clientId: String? = null

    var password: String? = null
    var period: String? = null

    var usesServer: PreferenceUtils.Server? = null

    fun checkCredentials(): Boolean {
        return !(password == null || password!!.isEmpty() || serverHost == null || serverHost!!.isEmpty())
    }

    fun usesNA(): Boolean {
        return usesServer == PreferenceUtils.Server.NA
    }

    fun usesEU(): Boolean {
        return usesServer == PreferenceUtils.Server.EU
    }
}
